pkg/apis/serving/v1alpha2/kfservice_status.go

```diff
@@ -122,30 +122,32 @@ func (ss *KFServiceStatus) PropagateCanaryStatus(endpoint constants.KFServiceEnd
 
 func (ss *KFServiceStatus) propagateStatus(statusSpec *StatusConfigurationSpec, conditionType apis.ConditionType, serviceStatus *knservingv1alpha1.ServiceStatus) {
 	statusSpec.Name = serviceStatus.LatestCreatedRevisionName
-	if serviceStatus.URL != nil {
-		statusSpec.Hostname = serviceStatus.URL.Host
-	}
 
 	serviceCondition := serviceStatus.GetCondition(knservingv1alpha1.ServiceConditionReady)
+
 	switch {
 	case serviceCondition == nil:
 	case serviceCondition.Status == v1.ConditionUnknown:
 		conditionSet.Manage(ss).MarkUnknown(conditionType, serviceCondition.Reason, serviceCondition.Message)
+		statusSpec.Hostname = ""
 	case serviceCondition.Status == v1.ConditionTrue:
 		conditionSet.Manage(ss).MarkTrue(conditionType)
+		if serviceStatus.URL != nil {
+			statusSpec.Hostname = serviceStatus.URL.Host
```

Comment:
While we're in here, should we refactor Hostname to Host? Can you remind me why we're not propagating the URL entirely?

Reply:
The Go URL struct does not have `openapi-gen` annotation, so it does not work with code generation and it contains a bunch of fields we do not need, we could make our own lean version of URL struct.

Reply:
The Knative addressable struct also calls this Hostname... so going to leave it for now..

```
type Addressable struct {
	v1beta1.Addressable `json:",omitempty"`

	Hostname string `json:"hostname,omitempty"`
}
```